drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/ScenarioWidgetComponentCreator.java

```diff
@@ -74,21 +74,26 @@ protected void reset(final ScenarioParentWidget scenarioWidget,
         this.ruleNamesService.call((Collection<String> ruleNames) -> {
             if (!(ruleNames == null || ruleNames.isEmpty())) {
                 ruleNameSelector.setEnabled(true);
-                ruleNames.forEach((r) -> ruleNameSelector.add(makeRuleNameOption(r)));
+                ruleNames.forEach((r) -> ruleNameSelector.add(makeRuleNameOption(getSimpleRuleName(r))));
             } else {
                 ruleNameSelector.setEnabled(false);
             }
         }).getRuleNames(path,
                         scenario.getPackageName());
     }
 
-    Option makeRuleNameOption(final String text) {
+    Option makeRuleNameOption(final String simpleRuleName) {
         final Option o = GWT.create(Option.class);
-        o.setText(text);
-        o.setValue(text);
+        o.setText(simpleRuleName);
+        o.setValue(simpleRuleName);
         return o;
     }
 
+    String getSimpleRuleName(final String fullyQualifiedRuleName) {
+        final int dotIndex = fullyQualifiedRuleName.lastIndexOf(".");
+        return dotIndex > -1 ? fullyQualifiedRuleName.substring(dotIndex + 1) : fullyQualifiedRuleName;
+    }
```

Comment:
@manstis Great point. So we've uncovered the root issue with the index, where FQRNs are ambiguous! For example, `com.somepgk.my.r1` can mean either

    package com.somepkg; rule "my.r1"
or

    package com.somepkg.my; rule "r1"

right?

Would it be feasible to remove the root cause by fixing the index so that it stores FQRNs in an unambiguous manner? For example, `com.mypkg."some rule with.dots;and;stuff?"` (rule simple name enclosed with double quotes).